spec/system/account_modal_system_spec.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2008-2017, Puzzle ITC GmbH. This file is part of
+#  Cryptopus and licensed under the Affero General Public License version 3 or later.
+#  See the COPYING file at the top-level directory or at
+#  https://github.com/puzzle/cryptopus.
+
+require 'rails_helper'
+
+
+describe 'AccountModal', type: :system, js: true do
+  include SystemHelpers
+
+  it 'creates, edits and deletes an account' do
+    login_as_user(:bob)
+
+    account_attr = { accountname: 'acc',
+                     username: 'username',
+                     password: 'password',
+                     description: 'desc' }
+
+    account_attr_edited = { accountname: 'acc2',
+                            username: 'username2',
+                            password: 'password2',
+                            description: 'desc2' }
+
+    # Create Account
+    expect(page).to have_link('new Account')
+    click_link 'new Account'
+
+    expect(find('.modal-content')).to be_present
+    expect(page).to have_text('New Account')
+    expect(page).to have_button('Save')
+
+    expect do
+      fill_modal(account_attr)
+      click_button 'Save'
+    end.to change { Account.count }.by(1)
+
+    expect_account_page_with(account_attr)
+
+    # Edit Account
+    account = Account.find_by(accountname: account_attr[:accountname])
+    group = Group.find_by(id: account.group_id)
+    team = Team.find_by(id: group.team_id)
+    visit("/accounts/#{account.id}")
+
+    expect(page).to have_link(id: 'edit_account_button')
+    click_link(id: 'edit_account_button')
+
+    expect(find('.modal-content')).to be_present
+    expect(page).to have_text('Edit Account')
+    expect(page).to have_button('Save')
+
+    expect_filled_fields_in_modal_with(account_attr)
+
+    fill_modal(account_attr_edited)
+    click_button 'Save'
+
+    expect_account_page_with(account_attr_edited)
+
+    # Delete Account
+    find(:xpath, "//a[@href='/teams/#{group.team_id}/groups/#{account.group_id}']").click
+    expect(find('h1')).to have_text("Accounts in group #{group.name} for team #{team.name}")
+
+    # Still an async issue. So sometimes test passes, sometimes not.
+    expect do
+      accept_prompt(wait: 10) do
+        del_button = find(:xpath, "//a[@href='/accounts/#{account.id}' and @data-method='delete']")
+        expect(del_button).to be_present
+      end
+    end.to change { Account.count }.by(-1)
```

Comment:
There is still an issue with asynchronity. I haven't found out, yet, why, as I wait 10 seconds max for the prompt message to delete the account.
So sometimes this lines fails and `Account.count` has not been changed by -1 after the deletion, as probably it couldn't delete, because it didn't wait long enough somewhere.